docs/UsingTheIOSubsystem.md

```diff
@@ -0,0 +1,615 @@
+- **Network information** (e.g., getaddrinfo, getnameinfo)
+- **Process/user/group information** (e.g., getpid, getuid, getgid, etc.)
+
+Chapter 1 provides an overview of the subsystem. Chapter 2 discusses which
+system functions are supported.
+
+Chapter 1: Overview
+===================
+
+The **Open Enclave I/O subsystem** exposes I/O features through ordinary system
+header files. These headers fall roughly into three categories.
+
+- **C headers** (such as stdio.h, stdlib.h)
+- **POSIX headers** (such as socket.h, dirent.h, fnctl.h, stat.h)
+- **Linux headers** (such as mount.h)
+
+**Open Enclave** provides a partial implementation of these headers and many
+others. These headers are part of [**musl libc**](https://www.musl-libc.org),
+which **Open Enclave** redistributes for use within enclaves under the name
+**oelibc**. The I/O subsystem expands **oelibc** to support many additional
+functions.
+
+For the most part, developers just use standard headers to build enclaves;
+however, a few additional functions are needed to select which I/O features
+are needed. This is purely a security measure. By default, enclaves have no
+access to files or sockets. The next section describes how to opt-in to
+various I/O features.
+
+Opting in
+---------
+
+An enclave application must first opt in to the I/O features it wishes to
+use. Opting in is a matter of (1) linking the desired module libraries and
+(2) calling functions to load the modules. I/O features are packaged as
+static libraries. This release provides the following modules.
+
+- **liboehostfs** -- access to non-secure host files and directories.
+- **liboehostsock** -- access to non-secure sockets.
+- **libhostresolver** -- access to network information.
+
+After linking modules, the enclave loads modules by calling one of the
+following.
+
+- **oe_load_module_host_file_system()**
+- **oe_load_module_host_socket_interface()**
+- **oe_load_module_host_resolver()**
+
+Operating system support
+------------------------
+
+The current version is Limited to Linux hosts but Windows host is under
+development now.
+
+File system path resolution
+---------------------------
+
+Path-oriented functions must resolve their path parameter to a file system
+device. There are two mechanisms for doing this. The first mechanisms employs
+the Linux **mount** function defined below.
+
+```
+#include <sys/mount.h>
+
+int mount(
+    const char* source,
+    const char* target,
+    const char* filesystemtype,
+    unsigned long mountflags,
+    const void* data);
+```
+
+This function attaches a file system to the directory specifies by **target**.
+**filesystemtype** parameter specifies the name of the file system. The next
+section will show how to mount the non-secure host file system.
+
+The second path-resolution mechanism employs the following functions.
+
+```
+oe_result_t oe_set_thread_devid(uint64_t devid);
+oe_result_t oe_clear_thread_devid();
```

Comment:
I am curious about why oe_set_thread_devid() and oe_clear_thread_devid() are mentioned here because 
they don't look like public APIs. Am I missing anything here?

Reply:
Yeah, I took it out. I was on the fence about that one. :)

Reply:
Thanks for addressing my comment.